src/app/data-planes/components/DataplanePolicies.vue

```diff
@@ -250,80 +252,103 @@ function getPolicyEntryConnections(policy: MatchedPolicyType, policyType: Policy
+ * Transforms `DataplaneRule` objects into rule entries which are going to be displayed in this view.
  */
-function getPolicyTypeEntriesFromRules(rules: DataplaneRule[]): PolicyTypeEntry[] {
+function getRuleEntries(rules: DataplaneRule[]): RuleEntry[] {
   // Uses a `Map` to store entries by type so they can be retrieved and updated while iterating over the rules.
-  const policyEntriesByType = new Map<string, PolicyTypeEntry>()
+  const policyTypeEntriesByType = new Map<string, RuleEntry>()
 
   for (const rule of rules) {
-    if (!policyEntriesByType.has(rule.policyType)) {
-      policyEntriesByType.set(rule.policyType, {
+    if (!policyTypeEntriesByType.has(rule.policyType)) {
+      policyTypeEntriesByType.set(rule.policyType, {
         type: rule.policyType,
         connections: [],
       })
     }
 
-    const policyEntry = policyEntriesByType.get(rule.policyType) as PolicyTypeEntry
+    const policyTypeEntry = policyTypeEntriesByType.get(rule.policyType) as RuleEntry
     const policyType = store.state.policyTypesByName[rule.policyType]
-    const connections = getPolicyEntryConnectionsFromRules(rule, policyType)
+    const connections = getRuleEntryConnections(rule, policyType)
 
-    policyEntry.connections.push(...connections)
+    policyTypeEntry.connections.push(...connections)
   }
 
-  const policyTypeEntries = Array.from(policyEntriesByType.values())
+  const policyTypeEntries = Array.from(policyTypeEntriesByType.values())
 
-  policyTypeEntries.sort((policyEntryA, policyEntryB) => policyEntryA.type.localeCompare(policyEntryB.type))
+  policyTypeEntries.sort((policyTypeEntryA, policyTypeEntryB) => policyTypeEntryA.type.localeCompare(policyTypeEntryB.type))
 
   return policyTypeEntries
 }
 
-function getPolicyEntryConnectionsFromRules(rule: DataplaneRule, policyType: PolicyType): PolicyTypeEntryConnection[] {
+function getRuleEntryConnections(rule: DataplaneRule, policyType: PolicyType): RuleEntryConnection[] {
   const { type, service, subset, conf } = rule
 
   // Guards against likely API changes. The response currently contains `"subset": {}` instead of omitting the value or setting it to `null`, but that might change in the future.
   const subsetEntries = subset ? Object.entries(subset) : []
   let sourceTags: LabelValue[] | null
   let destinationTags: LabelValue[] | null
 
+  // Determines source tags.
   if (type === 'clientSubset') {
+    // For client subsets, the source is represented by `subset` (i.e. tags)
     // Sets the wildcard service tag for client subsets without subset values.
     const sourceSubset = subsetEntries.length > 0 ? subsetEntries : [['kuma.io/service', '*']]
-    // For client subsets, the source is represented by `subset` (i.e. tags); for destination subsets and single items, the source is the DPP.
     sourceTags = sourceSubset.length > 0 ? sourceSubset.map(([label, value]) => ({ label, value })) : null
```

Comment:
I know this is from a previous PR but maybe a "whilst you are here", plus it would help with an overall opinion here that this function is a little hard to look at, and there are a few things that I'm looking at and wondering about things. Also it looks like this function is the main chunk of the PR (the rest looks like mostly mocks and variable renaming)

Anyway:

Isn't `sourceSubset` here always guaranteed to be `> 0` length?

i.e. If `subsetEntries` is `> 0` length, then we use `subsetEntries` for `sourceSubset` and hence `sourceSubset` is `> 0` length, and if `subsetEntries` isn't ` > 0` length then we set it to something that is `> 0` length and hence `sourceSubset` is `> 0` length.

Therefore `sourceSubset` is always `> 0` length?

I dunno maybe I read it wrong?

Reply:
You’re correct. In this branch, we can simplify the logic slightly. I’ve adopted the same explicit if statement branching as I’ve used below for setting `destinationTags`.